specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+| `TOKEN` |  `0x4200000000000000000000000000000000000042` | The address of the `GovernanceToken` contract  |
+
+## Interface
+
+### Core Functions
+
+#### `cancel`
+
+Cancels a proposal. A proposal MUST only be cancellable by the manager and ONLY while it's in an active state,
+i.e. not executed or cancelled already.
+
+```solidity
+function cancel(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `cancelWithModule`
+
+Similar to `cancel`, but for proposals that were created with a module.
+
+```solidity
+function cancelWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `castVote`
+
+Cast a vote on a proposal.
+
+```solidity
+function castVote(uint256 _proposalId, uint8 _support) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteBySig`
+
+Cast a vote on a proposal using a signature.
+
+```solidity
+function castVoteBySig(uint256 _proposalId, uint8 _support, uint8 _v, bytes32 _r, bytes32 _s) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteWithReason`
+
+Cast a vote on a proposal with a reason.
+
+```solidity
+function castVoteWithReason(uint256 _proposalId, uint8 _support, string memory _reason) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteWithReasonAndParams`
+
+Cast a vote on a proposal with a reason and additional encoded parameters.
+
+```solidity
+function castVoteWithReasonAndParams(uint256 _proposalId, uint8 _support, string memory _reason, bytes memory _params) external returns (uint256)
+```
+
+This function MUST emit the `VoteCastWithParams` event if the length of the parameters is more than zero. Otherwise,
+the function MUST emit the `VoteCast` event. The function MUST also return the voting weight.
+
+#### `castVoteWithReasonAndParamsBySig`
+
+Cast a vote on a proposal with a reason and additional encoded parameters using a signature.
+
+```solidity
+function castVoteWithReasonAndParams(uint256 _proposalId, uint8 _support, string memory _reason, bytes memory _params, uint8 _v, bytes32 _r, bytes32 _s) external returns (uint256)
+```
+
+This function MUST emit the `VoteCastWithParams` event if the length of the parameters is more than zero. Otherwise,
+the function MUST emit the `VoteCast` event. The function MUST also return the voting weight.
+
+#### `editProposalType`
+
+Edit the type of a proposal that is still active. This function MUST only be callable by the manager.
```

Comment:
What is a proposal type?

Reply:
improved the `Proposal Types` section and the description for this function so that they are now connected and better specified: https://github.com/ethereum-optimism/specs/pull/315/commits/495398c73b5e5c4825f64b99864e99976a69f673